DataDog/dd-trace-rb
# Misc small fixes to profiling native extension

**What does this PR do?**:

This PR compiles a bunch of small fixes and improvements to the profiling native extension. I recommend reviewing it commit-by-commit, every commit includes a clear description of what is being changed and why.

**Motivation**:

Started solving a few compiler warnings and discovered a bunch of things to be improved.

**How to test the change?**:

Confirm that CI is still green -- none of these changes is expected to change behavior in any way that matters.
